src/Confluent.Kafka/AdminClient.cs

```diff
@@ -0,0 +1,750 @@
+        public AdminClient(Handle handle)
+        {
+            this.ownedClient = null;
+            this.handle = handle;
+            Init();
+        }
+
+        private void Init()
+        {
+            // note: ConcurrentDictionary.GetorAdd() method is not atomic
+            this.topicHandlerFactory = (string topicName) =>
+            {
+                // Note: there is a possible (benign) race condition here - topicHandle could have already
+                // been created for the topic (and possibly added to topicHandles). If the topicHandle has
+                // already been created, rdkafka will return it and not create another. the call to rdkafka
+                // is threadsafe.
+                return kafkaHandle.Topic(topicName, IntPtr.Zero);
+            };
+
+            resultQueue = kafkaHandle.CreateQueue();
+
+            callbackCts = new CancellationTokenSource();
+            callbackTask = StartPollTask(callbackCts.Token);
+        }
+
+
+        /// <summary>
+        ///     Get information pertaining to all groups in the Kafka cluster (blocking)
+        ///
+        ///     [API-SUBJECT-TO-CHANGE] - The API associated with this functionality 
+        ///     is subject to change.
+        /// </summary>
+        /// <param name="timeout">
+        ///     The maximum period of time the call may block.
+        /// </param>
+        public List<GroupInfo> ListGroups(TimeSpan timeout)
+            => kafkaHandle.ListGroups(timeout.TotalMillisecondsAsInt());
+
+
+        /// <summary>
+        ///     Get information pertaining to a particular group in the
+        ///     Kafka cluster (blocking).
+        ///
+        ///     [API-SUBJECT-TO-CHANGE] - The API associated with this functionality is subject to change.
+        /// </summary>
+        /// <param name="group">
+        ///     The group of interest.
+        /// </param>
+        /// <param name="timeout">
+        ///     The maximum period of time the call may block.
+        /// </param>
+        /// <returns>
+        ///     Returns information pertaining to the specified group
+        ///     or null if this group does not exist.
+        /// </returns>
+        public GroupInfo ListGroup(string group, TimeSpan timeout)
+            => kafkaHandle.ListGroup(group, timeout.TotalMillisecondsAsInt());
+
+
+        ///  <summary>
+        ///     Get information pertaining to a particular group in the
+        ///     Kafka cluster (blocks, potentially indefinitely).
+        ///
+        ///     [API-SUBJECT-TO-CHANGE] - The API associated with this functionality is subject to change.
+        /// </summary>
+        /// <param name="group">
+        ///     The group of interest.
+        /// </param>
+        /// <returns>
+        ///     Returns information pertaining to the specified group
+        ///     or null if this group does not exist.
+        /// </returns>
+        public GroupInfo ListGroup(string group)
```

Comment:
Yep (you're telling me back what I've been telling you!). All blocking methods should ideally also have an optional `CancellationToken`. The reason this method hasn't been removed is just that I've been ignoring these admin-type methods since they're going to be depreciated in favor of better ones. I'll remove this now.